app/services/katello/candlepin/server.rb

```diff
@@ -0,0 +1,20 @@
+module Katello
+  module Candlepin
+    class Server
+      def self.init
+        return unless SETTINGS[:katello][:use_candlepin] && !Foreman.in_rake?('db:migrate')
+        candlepin_status = Katello::Ping.ping([:candlepin])
+
+        if candlepin_status[:status] == Katello::Ping::FAIL_RETURN_CODE
+          fail _("There was an issue with the Candlepin service %s: ") % candlepin_status[:candlepin][:message]
+        end
+
+        Organization.all.each do |org|
+          User.current = User.anonymous_admin
+          ForemanTasks.sync_task(::Actions::Katello::Organization::Create, org) unless org.library
```

Comment:
Foreman tasks can be async or sync, this one is doing it synchronously to ensure the creates happen before moving on. If we do this async, the user faces potential issues if the task hasn't finished but they are aloud to work on other entities such as an activation key.